src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -216,6 +221,8 @@ private async Task InitializeAsync()
             _solutionSaveAsEvent.AfterExecute += SolutionSaveAs_AfterExecute;
 
             _projectSystemCache.CacheUpdated += NuGetCacheUpdate_After;
+
+            NuGetPowerShellUsageCollector nuGetPowerShellUsageCollectorInstance = await _asyncServiceProvider.GetServiceAsync<NuGetPowerShellUsageCollector>();
```

Comment:
It's funky thing added because if I don't create instance of `NuGetPowerShellUsageCollector` before event subscription `NuGetPowerShellUsage.RaiseSolutionOpenEvent();` then actual notification is not working. Can you recommend better way of doing it?

Reply:
This one addressed.